opentelemetry-sdk/src/metrics/data/mod.rs

```diff
@@ -38,23 +38,77 @@ pub struct Metric {
     /// The unit in which the instrument reports.
     pub unit: Cow<'static, str>,
     /// The aggregated data from an instrument.
-    pub data: Box<dyn Aggregation>,
+    pub data: AggregatedMetrics,
 }
 
-/// The store of data reported by an [Instrument].
-///
-/// It will be one of: [Gauge], [Sum], or [Histogram].
-///
-/// [Instrument]: crate::metrics::Instrument
-pub trait Aggregation: fmt::Debug + any::Any + Send + Sync {
-    /// Support downcasting
-    fn as_any(&self) -> &dyn any::Any;
-    /// Support downcasting during aggregation
-    fn as_mut(&mut self) -> &mut dyn any::Any;
+/// Aggregated metrics data from an instrument
+#[derive(Debug, Clone)]
```

Comment:
See if you like this better:
https://github.com/open-telemetry/opentelemetry-rust/compare/main...cijothomas:opentelemetry-rust:cijothomas/new-metrics#diff-041ac5cb4c3201f64e52be34cdba07e769e2b3637d605ddf59852a66ef104762R7-R70